File: LgfCommon/commonlibrary/src/main/java/com/cn/lgf/common/base/BaseApplication.java
package com.cn.lgf.common.base;

import android.app.Application;
import androidx.annotation.NonNull;
import androidx.lifecycle.ViewModelStore;
import androidx.lifecycle.ViewModelStoreOwner;
import org.jetbrains.annotations.NotNull;

/**
 * @ProjectName: LgfCommon
 * @Package: com.cn.lgf.common.base
 * @ClassName: BaseApplication
 * @Description: Application基类
 * @Author: 作者名
 * @CreateDate: 2020/7/31 10:08 上午
 * @UpdateUser: 更新者：
 * @UpdateDate: 2020/7/31 10:08 上午
 * @UpdateRemark: 更新说明：
 * @Version: 1.0
 */
public class BaseApplication extends Application implements ViewModelStoreOwner {

    private ViewModelStore mAppViewModelStore;
    private static BaseApplication sInstance;

    @Override
    public void onCreate() {
        super.onCreate();
        sInstance = this;
        mAppViewModelStore = new ViewModelStore();
    }

    @NonNull
    @NotNull
    @Override
    public ViewModelStore getViewModelStore() {
        return mAppViewModelStore;
    }

    public static BaseApplication getInstance() {
        return sInstance;
    }
}

File: LgfCommon/commonlibrary/src/main/java/com/cn/lgf/common/BaseApplication.kt
package com.cn.lgf.common

import android.app.Application
import android.content.Context

/**
 *
 * @ProjectName:    LgfCommon
 * @Package:        com.cn.lgf.common
 * @ClassName:      MyApp
 * @Description:     java类作用描述
 * @Author:         作者名
 * @CreateDate:     2020/7/13 12:52 下午
 * @UpdateUser:     更新者：
 * @UpdateDate:     2020/7/13 12:52 下午
 * @UpdateRemark:   更新说明：
 * @Version:        1.0
 */
open class BaseApplication() : Application() {
    companion object {
        private var sInstance: BaseApplication? = null
        fun getInstance(): BaseApplication? {
            return sInstance;
        }

    }

    override fun attachBaseContext(base: Context?) {
        super.attachBaseContext(base)
    }

    override fun onCreate() {
        super.onCreate()
        sInstance = this
    }
}